bin/check-rabbitmq-node-usage.rb

```diff
@@ -0,0 +1,306 @@
+         description: 'Resource type',
+         required: true,
+         long: '--type TYPE',
+         in: %w(mem socket fd proc disk)
+
+  option :memwarn,
+         description: 'Warning % of mem usage vs high watermark',
+         short: '-m',
+         long: '--mwarn PERCENT',
+         proc: proc(&:to_f),
+         default: 80
+
+  option :memcrit,
+         description: 'Critical % of mem usage vs high watermark',
+         short: '-c',
+         long: '--mcrit PERCENT',
+         proc: proc(&:to_f),
+         default: 90
+
+  option :fdwarn,
+         description: 'Warning % of file descriptor usage vs high watermark',
+         short: '-f',
+         long: '--fwarn PERCENT',
+         proc: proc(&:to_f),
+         default: 80
+
+  option :fdcrit,
+         description: 'Critical % of file descriptor usage vs high watermark',
+         short: '-F',
+         long: '--fcrit PERCENT',
+         proc: proc(&:to_f),
+         default: 90
+
+  option :socketwarn,
+         description: 'Warning % of socket usage vs high watermark',
+         short: '-s',
+         long: '--swarn PERCENT',
+         proc: proc(&:to_f),
+         default: 80
+
+  option :socketcrit,
+         description: 'Critical % of socket usage vs high watermark',
+         short: '-S',
+         long: '--scrit PERCENT',
+         proc: proc(&:to_f),
+         default: 90
+
+  option :procwarn,
+         description: 'Warning % of proc usage vs high watermark',
+         short: '-e',
+         long: '--pwarn PERCENT',
+         proc: proc(&:to_f),
+         default: 80
+
+  option :proccrit,
+         description: 'Critical % of proc usage vs high watermark',
+         short: '-E',
+         long: '--pcrit PERCENT',
+         proc: proc(&:to_f),
+         default: 90
+
+  option :diskwarn,
+         description: 'Warning % of disk usage vs high watermark',
+         short: '-d',
+         long: '--dwarn PERCENT',
+         proc: proc(&:to_f),
+         default: 80
+
+  option :diskcrit,
+         description: 'Critical % of disk usage vs high watermark',
+         short: '-D',
+         long: '--dcrit PERCENT',
+         proc: proc(&:to_f),
+         default: 90
+
+  def run
+    res = node_healthy?
+
+    if res['status'] == 'ok'
+      ok res['message']
+    elsif res['status'] == 'warning'
+      warning res['message']
+    elsif res['status'] == 'critical'
+      critical res['message']
+    else
+      unknown res['message']
+    end
+  end
+
+  def node_healthy?
+    # Checks and shows usage (for the specified type only). Shows usage for ok status as well.
+    case config[:type]
+    when 'mem'
+      output = node_mem
+    when 'socket'
+      output = node_socket
+    when 'fd'
+      output = node_fd
+    when 'proc'
+      output = node_proc
+    when 'disk'
+      output = node_disk
+    end
+
+    { 'status' => output[:status], 'message' => output[:message] }
+  rescue Errno::ECONNREFUSED => e
+    { 'status' => 'critical', 'message' => e.message }
```

Comment:
we already have checks to determine if a node is alive `check-rabbitmq-alive.rb`, this should be an unknown to reduce the number of parallel alerts.